src/routes/email/email.controller.update-email.spec.ts

```diff
@@ -0,0 +1,212 @@
+import { Test, TestingModule } from '@nestjs/testing';
+import { AppModule } from '@/app.module';
+import { CacheModule } from '@/datasources/cache/cache.module';
+import { TestCacheModule } from '@/datasources/cache/__tests__/test.cache.module';
+import configuration from '@/config/entities/__tests__/configuration';
+import { RequestScopedLoggingModule } from '@/logging/logging.module';
+import { TestLoggingModule } from '@/logging/__tests__/test.logging.module';
+import { NetworkModule } from '@/datasources/network/network.module';
+import { TestNetworkModule } from '@/datasources/network/__tests__/test.network.module';
+import { TestAppProvider } from '@/__tests__/test-app.provider';
+import { EmailDataSourceModule } from '@/datasources/email/email.datasource.module';
+import { TestEmailDatasourceModule } from '@/datasources/email/__tests__/test.email.datasource.module';
+import * as request from 'supertest';
+import { faker } from '@faker-js/faker';
+import { generatePrivateKey, privateKeyToAccount } from 'viem/accounts';
+import { IConfigurationService } from '@/config/configuration.service.interface';
+import { NetworkService } from '@/datasources/network/network.service.interface';
+import { IEmailDataSource } from '@/domain/interfaces/email.datasource.interface';
+import { chainBuilder } from '@/domain/chains/entities/__tests__/chain.builder';
+import { safeBuilder } from '@/domain/safe/entities/__tests__/safe.builder';
+import { getAddress } from 'viem';
+import { EmailControllerModule } from '@/routes/email/email.controller.module';
+
+describe('Email controller update email tests', () => {
```

Comment:
It looks like we are missing a test which tests the expected response if an `EmailUpdateError` is thrown.

Reply:
Added in 234b4ce38efb7be32c9940dff56dcc943e511a90.

Reply:
It's still missing right? 🤔 `EmailUpdateError` in this case should return a `500` or similar.

Reply:
I initially thought the previously added test covered this. I've removed the `EmailUpdateError` as it was wrapping not only updates but also email verification errors. Either [throw and cause a `500`](https://github.com/safe-global/safe-client-gateway/blob/5674ff88d2bdb104e784280af3968346b9446d87/src/routes/email/email.controller.update-email.spec.ts#L203) as of 5674ff88d2bdb104e784280af3968346b9446d87.